src/Rings/MPolyMap/flattenings.jl

```diff
@@ -421,30 +203,82 @@ end
 end
 
 ### Deflecting some of the basic methods for ideals in towers of polynomial rings
-@attr function _flat_counterpart(
-    I::MPolyIdeal{RingElemType}
-  ) where { RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
+function ideal_membership(
+    x::AbstractAlgebra.Generic.MPoly{RingElemType},
+    I::MPolyIdeal{AbstractAlgebra.Generic.MPoly{RingElemType}}
+  ) where {RingElemType <: Union{<:MPolyElem, <:MPolyQuoElem, <:MPolyLocalizedRingElem, 
                                   <:MPolyQuoLocalizedRingElem
                                  }
           }
-  S = base_ring(I)
-  S_flat = _flatten(S)
-  return ideal(S_flat, _orig_to_flat(S).(gens(I)))
+  parent(x) === base_ring(I) || return base_ring(I)(x) in I
+  phi = flatten(parent(x))
+  return ideal_membership(phi(x), phi(I))
 end
 
-function ideal_membership(
+function coordinates(
```

Comment:
I don't seen any example, so can you explain what this does? I am bit confused. More precisely, if `v = coordinates(x, I)`, does `x = sum(x[i] * gens(I)[i])` hold?

Reply:
Yes, `x = sum(v[i]*gens(I, i))` holds, exactly. The `coordinates` function is also implemented for ideals over polynomial rings, quotient rings, and their localizations.

Reply:
I was confused for a moment and thought that `coefficients` for things in `R[x]` will be in `R`, but this here is about the ring structure and not the `R`-algebra structure.  Makes sense! Thanks.